types/mongodb/test/collection/updateX.ts

```diff
@@ -254,3 +255,26 @@ async function run() {
         },
     );
 }
+
+async function testPushWithId () {
+    interface Model {
+        _id: ObjectId;
+        foo: Array<{ _id?: string, name: string }>;
+    }
+
+    const client = await connect(connectionString);
+    const db = client.db('test');
+    const collection = db.collection<Model>('test');
+
+    await collection.updateOne({}, {
+        $push: {
+            foo: { name: 'Foo' }
+        }
+    });
+
+    await collection.updateOne({}, {
+        $push: {
+            foo: { _id: 'foo', name: 'Foo' }
```

Comment:
This is the way that I first noticed this issue. I had a schema similar to this one, where the `_id` inside the array was typed as `_id?: string`. When I tried to `$push` to the array I got an error that `_id` wasn't an `ObjectId` 😄